src/ngraph/runtime/cpu/builder/lstm.cpp

```diff
@@ -56,20 +55,52 @@ namespace ngraph
                 auto& dst_iter_tensor = external_function->get_tensor_data(out[1].get_name());
 
                 auto& mkldnn_emitter = external_function->get_mkldnn_emitter();
-                auto lstm_index = mkldnn_emitter->build_rnn<ngraph::op::Lstm>(node, args, out);
-                auto& deps = mkldnn_emitter->get_primitive_deps(lstm_index);
+                auto index = mkldnn_emitter->build_rnn<ngraph::op::Lstm>(node, args, out);
+                if (index.size() != 3)
+                {
+                    throw ngraph_error(
+                        "Rnn builder doesnot have required primitives: {weight_layer_reorder, "
+                        "weight_iter_reorder, rnn_primtive}");
+                }
+                auto& rnn_deps = mkldnn_emitter->get_primitive_deps(index[2]);
+                auto& weights_layer_deps = mkldnn_emitter->get_primitive_deps(index[0]);
+                auto& weights_iter_deps = mkldnn_emitter->get_primitive_deps(index[1]);
 
-                auto functor = [&, lstm_index](CPURuntimeContext* ctx) {
-                    cpu::mkldnn_utils::set_memory_ptr(ctx, deps[0], src_layer_tensor);
-                    cpu::mkldnn_utils::set_memory_ptr(ctx, deps[1], src_iter_tensor);
-                    cpu::mkldnn_utils::set_memory_ptr(ctx, deps[2], weights_layer_tensor);
-                    cpu::mkldnn_utils::set_memory_ptr(ctx, deps[3], weights_iter_tensor);
-                    cpu::mkldnn_utils::set_memory_ptr(ctx, deps[4], bias_tensor);
-                    cpu::mkldnn_utils::set_memory_ptr(ctx, deps[5], dst_layer_tensor);
-                    cpu::mkldnn_utils::set_memory_ptr(ctx, deps[6], dst_iter_tensor);
+                auto functor_weights_layer_reorder = [&, index](CPURuntimeContext* ctx) {
```

Comment:
Can we do the reorder using reshape nodes instead of hiding it inside the RNN op. This will enable computation caching to kick in and skip reorders on inference. You might have to change the fusion patterns.